Makefile

```diff
@@ -44,14 +52,31 @@ export GOARCH ?= amd64
 
 
 # @todo allow user to run for a single $PKG only?
-PACKAGES := $(shell $(GO) list ./...)
+PACKAGES_CMD := $(GO) list ./...
 GO_EXCLUDE := /vendor/|.pb.go|.gen.go
-GO_FILES := $(shell find . -name '*.go' | grep -v -E '$(GO_EXCLUDE)')
+GO_FILES_CMD := find . -name '*.go' | grep -v -E '$(GO_EXCLUDE)'
 
 # Environment for tests, the directory containing istio and deps binaries.
 # Typically same as GOPATH/bin, so tests work seemlessly with IDEs.
 export ISTIO_BIN=${GO_TOP}/bin
 
+# go version that's major # * 100 + minor #
+# if you change this value then also update the echo statement in the section that follows
+GO_VERSION_REQUIRED=109
```

Comment:
it may be worth taking that section and moving it into a shell script that would deal with the more natural "1.9.1" and can be reused ? (I guess go dep has that logic too)

Reply:
The impression I had in informal chats is that in the long run there may only be the top-level makefile, so I'm not sure there will be anything left to reuse it.  Overall I was hoping to reduce the number of script files that are used by istio.

I can try harder to make GO_VERSION_REQUIRED handle a string like "1.9" rather than "109".  I'm thinking I can do it by replacing the right side of the "-lt" comparison with something that's very much like the left side (except replace the "shell go version" with "echo $(GO_VERSION_REQUIRED)").  I know I can probably get awk also do sed's regex task but I've hardly ever used awk so it'll likely take me longer to figure out than other people.

Reply:
I also want to point out it's likely someday we find a bug that istio triggers in say go 1.10.1 and we thus require >= 1.10.2

I'm sure parsing a sem ver 3 digit version and comparing is code that exists somewhere already

Reply:
I added a revision so that awk handles the task of sed|awk, 3 version components can be handled, and now GO_VERSION_REQUIRED can be expressed in a natural form like 1.9 or 1.9.2 (instead of 109).  I replaced the "echo -n" with printf since the "-n" was being echo'ed on Mac (the echo manpage on Mac suggested using printf -- that page says that "-n" is supported [and it does seem to work in simple cases, but not here for some reason]).